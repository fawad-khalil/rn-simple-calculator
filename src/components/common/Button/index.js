import React from 'react';

import { TouchableOpacity, Text } from 'react-native';
import styles from './styles';

const Button = ({ testID, buttonStyle, contentStyle, children, disabled, pressHandlerParams, onPress }) => {
  return (
    <TouchableOpacity
      testID={testID}
      style={{
        ...styles.button,
        ...buttonStyle,
      }}
      disabled={disabled}
      onPress={() => onPress({ ...pressHandlerParams })}
    >
      <Text style={{ ...contentStyle }}>{children}</Text>
    </TouchableOpacity>
  );
};

Button.defaultProps = {
  buttonStyle: {},
  contentStyle: {},
  pressHandlerParams: {},
  onPress: () => null,
};

export default Button;
