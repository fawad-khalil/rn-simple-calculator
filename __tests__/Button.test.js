/**
 * @format
 */

/**
 * @jest-environment jsdom
 */

// import 'jsdom-global/register';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure, ReactWrapper } from 'enzyme';
import Button from '../src/components/common/Button';

describe('<Button />', () => {
  describe('<Button />', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(
          <Button testID="testButton" onPress={jest.fn()}>
            Test Button
          </Button>,
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
