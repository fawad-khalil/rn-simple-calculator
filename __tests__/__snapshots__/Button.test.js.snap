// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="testButton"
>
  <Text
    style={Object {}}
  >
    Test Button
  </Text>
</View>
`;
